File: crates/ariadne-parser/fixtures/rust/sample.rs
use std::collections::HashMap;

pub struct Counter {
    pub value: u32,
}

pub enum Tick {
    Up,
    Down,
}

pub trait Step {
    fn step(&mut self, t: Tick);
}

pub type Bag = HashMap<String, u32>;

impl Counter {
    pub fn new(start: u32) -> Self {
        Self { value: start }
    }

    pub fn increment(&mut self) -> u32 {
        self.value += 1;
        self.value
    }
}

pub fn fresh() -> Counter {
    Counter::new(0)
}

mod inner {
    pub fn ping() -> u32 {
        42
    }
}

File: crates/ariadne-parser/fixtures/c/sample.c
#include <stddef.h>

struct Counter {
    unsigned int value;
};

enum Tick {
    TICK_UP,
    TICK_DOWN,
};

typedef struct Counter Counter;

unsigned int counter_increment(struct Counter *c) {
    c->value += 1;
    return c->value;
}

unsigned int counter_double(struct Counter *c) {
    return counter_increment(c) + counter_increment(c);
}
